src/platform/packages/shared/kbn-esql-utils/src/utils/get_initial_esql_query.ts

```diff
@@ -8,19 +8,44 @@
  */
 
 import type { DataView } from '@kbn/data-views-plugin/public';
+import type { Query } from '@kbn/es-query';
+
+const getFilterBySearchText = (query?: Query) => {
+  if (!query) {
+    return '';
+  }
+  const searchTextFunc =
+    query.language === 'kuery' ? 'KQL' : query.language === 'lucene' ? 'QSTR' : '';
+
+  if (searchTextFunc && query.query) {
+    return `${searchTextFunc}("""${query.query}""")`;
```

Comment:
It looks like the triple quotes take care of most escaping issues, but we might still want to escape double quotes since some valid Lucene queries can cause ES|QL parsing errors:

https://github.com/user-attachments/assets/048c68de-d5bf-4654-bd05-8cbae6ccd19e

I couldn't reproduce it with KQL except by running an invalid KQL query like `invalid """ query`, but escaping double quotes would take care of these cases too.

We have an existing `escapeQuotes` util we could export from `src/platform/packages/shared/kbn-es-query/src/kuery/utils/escape_kuery.ts` that I think should be safe for both Lucene and KQL queries.

Reply:
The usage of the triple quotes is important for training reasons. I dont want the users to get scared with escaping and this function unfortunately escapes the single quotes too. I am going to escape them in lucene but not in KQL https://github.com/elastic/kibana/pull/206391/commits/5afbf7ff296b604f1de9b1588317dd06ccc0eb61

Reply:
I didn't suggest moving away from the triple quotes, I actually think they're good since they avoid most escaping issues. I don't share the concerns about escaping single quotes since it works fine and is common when working with strings, but I think addressing only Lucene is ok too since it's where the valid queries can actually fail 👍